### PR TITLE
Begin fixing dialyzer

### PR DESCRIPTION
**Pull request by ConnorRigby**
While looking around to add a feature, I noticed that

1) dialyzer isn't run in CI
2) the existing specs were mostly wrong
3) there aren't very many existing specs

### CODE REVIEW

**Comment by fhunleth** [comment]
Thank you for adding some docs and specs!!!!

I think this is a pretty involved endeavor and would rather have some changes than all at once. I can't tell if the test error is due to the changes here or not. However, I'd be super happy to even have small steps. For example, enable Dialyzer, but don't fail CI if it fails. Then merge all of your doc updates. Then merge the easy Dialyzer fixes, etc.

@jjcarstens @mobileoverlord Are you two ok with this?

@ConnorRigby I see that you marked it as draft, but would you be comfortable getting us started with the really low risk changes?

**Comment by ConnorRigby** [comment]
>  I see that you marked it as draft, but would you be comfortable getting us started with the really low risk changes?
yep i'm fine with that. I was hoping to at least get the existing failures cleaned up, but the CI change can be reverted to allow for these incremental updates to be applied.

File: apps/nerves_hub_web_core/lib/nerves_hub_web_core/firmwares/firmware_metadata.ex [hunk 1]
```diff
@@ -21,6 +19,25 @@ defmodule NervesHubWebCore.Firmwares.FirmwareMetadata do
...
+  @typedoc """
+  fwup meta data about firmware
+
+  [read more here](https://github.com/fwup-home/fwup#global-scope)
+  """
+  @type t :: %__MODULE__{
+          architecture: String.t(),
+          author: String.t() | nil,
+          description: String.t() | nil,
+          fwup_version: Version.build(),
```
Comment (fhunleth) [inline apps/nerves_hub_web_core/lib/nerves_hub_web_core/firmwares/firmware_metadata.ex:31]:
```suggestion
          fwup_version: Version.build() | nil,
```

**Review by fhunleth — APPROVED** [review]

**Review by mobileoverlord — APPROVED** [review]
Looks good to me.

**Pull request merged**